Thorium-Sim/thorium-nova
# Ship map nodes frontend

## Description

Builds out the frontend for building nodes and edges on the ship map.

## Related Issue








Closes #165
Closes #167 
Closes #168 
Closes #169 

## How do you know the changes work correctly?






Manual testing
- Go to the ship editor
- Go to the ship map section
- Upload a ship map, if there isn't one already
- Choose "Add Nodes"
- Click to add a node
- Drag it around
- Change the properties by clicking to select
- Create another node
- Click "Add Edges"
- Click between two nodes to create an edge
- Click to select the edge and delete it.
- Create nodes on two separate decks and create an edge between them. There should be a second circle around the node with the cross-deck connection

## Screenshots (if appropriate):
<img width="1036" alt="Screen Shot 2022-03-12 at 7 49 55 PM" src="https://user-images.githubusercontent.com/6558157/158040160-c2bdfe8b-8513-40bf-b35c-832f267116e0.png">


## Types of changes



      change)

## Checklist:
